### PR TITLE
Fix search and add counts to meshmodel APIs

### LINKED ISSUE #7651: Enhance Top-Level API Endpoints with Count, Search, and Pagination Functionality for Improved User Experience
#### Current Behavior
We need top-level api endpoint to be able to search resources, paginate the results, and aggregate resources. 
For eg. `/api/meshmodels/models` do not return the models count, and search for models particularly. 

#### Desired Behavior

All the top-level endpoints need to return resource counts, allow resource to be searched and results be paginated. 

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by abdullah1308**
**Notes for Reviewers**

This PR fixes #7651

Added count to `/api/meshmodels/components`

image

Added count to `/api/meshmodels/models`

image

Added count to `/api/meshmodels/relationships`

image

Fixed search in `/api/meshmodels/models`

image

Fixed search in `/api/meshmodels/components`

image

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by abdullah1308** [comment]
@UsmanDev09 Let me know if any other changes are required in these or other endpoints

**Comment by abdullah1308** [comment]
Any suggestions on how to fix the failed tests ?

**Comment by leecalcote** [comment]
> Any suggestions on how to fix the failed tests ?

Tagging @MarioArriaga92

**Comment by Azanul** [comment]
@abdullah1308 code looks fine, good job. One improvement would be adding code comments/function descriptions to the functions that have been updated to return total count.

File: server/router/server.go [hunk 1]
```diff
@@ -154,7 +154,6 @@ func NewRouter(_ context.Context, h models.HandlerInterface, port int, g http.Ha
 	gMux.Handle("/api/meshmodels/categories", h.ProviderMiddleware(h.AuthMiddleware(http.HandlerFunc(h.GetMeshmodelCategories), models.NoAuth))).Methods("GET")
 	gMux.Handle("/api/meshmodels/models", h.ProviderMiddleware(h.AuthMiddleware(http.HandlerFunc(h.GetMeshmodelModels), models.NoAuth))).Methods("GET")
 	gMux.Handle("/api/meshmodels/models/{model}", h.ProviderMiddleware(h.AuthMiddleware(http.HandlerFunc(h.GetMeshmodelModelsByName), models.NoAuth))).Methods("GET")
-	gMux.Handle("/api/meshmodels/components", h.ProviderMiddleware(h.AuthMiddleware(http.HandlerFunc(h.GetAllMeshmodelRelationships), models.NoAuth))).Methods("GET")
```
Comment (MUzairS15) [inline server/router/server.go:157]: @abdullah1308 Is this intentional?

Comment (abdullah1308) [reply server/router/server.go:157]: @MUzairS15 yes as the following endpoint exists causing the above line to do nothing -

https://github.com/meshery/meshery/blob/051d838aadc60edcd6bed1bf9f0271b3fd07b126/server/router/server.go#L155

This endpoint has the exact same handlers and is functional -

https://github.com/meshery/meshery/blob/051d838aadc60edcd6bed1bf9f0271b3fd07b126/server/router/server.go#L172

**Comment by abdullah1308** [comment]
> @abdullah1308 code looks fine, good job. One improvement would be adding code comments/function descriptions to the functions that have been updated to return total count.

@Azanul done

**Comment by Azanul** [comment]
> > @abdullah1308 code looks fine, good job. One improvement would be adding code comments/function descriptions to the functions that have been updated to return total count.
> 
> @Azanul done

If you are mentioning `total_count` in swagger doc as you have, I think you also need to add a reference in `helpers/swagger.yaml`. Also, pascal case is used everywhere for it, so please use the same for consistency.

**Review by Abhishek-kumar09 — COMMENTED** [review]
🔴 Given it is API response restructuring, this should also require changes in other parts of the Meshery-UI, and Meshery-Extensions.

**Review by leecalcote — COMMENTED** [review]
@Revolyssup will you review?

**Comment by Chadha93** [comment]
@abdullah1308 Let's discuss this on dev call today, please add this as an agenda in meeting minutes if you would. - https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#heading=h.uvhhzvfx731s

**Comment by abdullah1308** [comment]
The PR https://github.com/meshery/meshkit/pull/316 needs to be closed to pass golang unit tests

**Comment by abdullah1308** [comment]
@Azanul Swagger docs have been fixed

**Comment by Abhishek-kumar09** [comment]
few of them (probably 1 or 2) is used in Meshery-Extension.

**Comment by Abhishek-kumar09** [comment]
@harkiratsm Would you assist the UI changes in Meshery-extension?

**Comment by UsmanDev09** [comment]
@Abhishek-kumar09 we are good to go. We are not using the endpoints mentioned by @abdullah1308. 

Graphql queries are being utilized where information from meshmodel is required.

**Comment by Azanul** [comment]
This PR can be merged after checking lint tests after merging https://github.com/meshery/meshkit/pull/316.
I think even without respective UI changes this won't break Meshery as the keys being used by UI are unchanged.

**Review by Abhishek-kumar09 — COMMENTED** [review]
@abdullah1308 @UsmanDev09 I just [released the Meshkit](https://github.com/meshery/meshkit/releases/tag/v0.6.45) with your changes, would you update go mod now.

**Review by Azanul — APPROVED** [review]

**Comment by leecalcote** [comment]
> > @abdullah1308 @UsmanDev09 I just [released the Meshkit](https://github.com/meshery/meshkit/releases/tag/v0.6.45) with your changes, would you update go mod now.
> 
> @Abhishek-kumar09 Done. Integration tests failing unfortunately.

@Abhishek-kumar09 what other areas need to be updated based on this change? Please list them out so that people can work through them.

**Comment by Abhishek-kumar09** [comment]
> > > @abdullah1308 @UsmanDev09 I just [released the Meshkit](https://github.com/meshery/meshkit/releases/tag/v0.6.45) with your changes, would you update go mod now.
> > 
> > 
> > @Abhishek-kumar09 Done. Integration tests failing unfortunately.
> 
> @Abhishek-kumar09 what other areas need to be updated based on this change? Please list them out so that people can work through them.

Listed it multiple times on call, assigned people and asked for the updates (multiple times through calls), but I don't think there is any success yet.

**Comment by Abhishek-kumar09** [comment]
> @Abhishek-kumar09 I checked meshery-extensions too. We are not using these api's

Then good to go

**Review by UsmanDev09 — APPROVED** [review]

**Pull request merged**